packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,286 @@
+            }
+        }
+    }
+
+    private checkConstructorParameters(construct: ts.ConstructorDeclaration): void {
+        switch (this.options.mode) {
+            case 'never':
+                for (const param of construct.parameters.filter(isParameterProperty))
+                    this.addFailureAtNode(
+                        param,
+                        FAILURE_STRING.never,
+                        getFixerForDisallowedParameterProp(
+                            construct,
+                            param,
+                            getLineBreakStyle(this.context.sourceFile),
+                        ),
+                    );
+                break;
+            case 'when-possible':
+                for (const param of construct.parameters)
+                    if (!isParameterProperty(param) && this.canBeParameterProperty(param, construct))
+                        this.addFailureAtNode(
+                            param,
+                            FAILURE_STRING.whenPossible,
+                            getFixerForLonghandProp(param, construct),
+                        );
+                break;
+            default:
+                if (construct.parameters.every(isParameterProperty)) return;
+
+                const allPropsCanBeParamProps = construct.parameters
+                    .filter((param) => !isParameterProperty(param))
+                    .every((param) => this.canBeParameterProperty(param, construct));
+
+                if (!allPropsCanBeParamProps && construct.parameters.some(isParameterProperty)) {
+                    for (const param of construct.parameters.filter(isParameterProperty))
+                        this.addFailure(
+                            construct.parameters.pos,
+                            construct.parameters.end,
+                            FAILURE_STRING.consistent.cannotBeParamPropsOnly,
+                            getFixerForDisallowedParameterProp(
+                                construct,
+                                param,
+                                getLineBreakStyle(this.context.sourceFile),
+                            ),
+                        );
+                } else if (allPropsCanBeParamProps && !construct.parameters.every(isParameterProperty)) {
+                    for (const param of construct.parameters.filter((p) => !isParameterProperty(p)))
+                        this.addFailure(
+                            construct.parameters.pos,
+                            construct.parameters.end,
+                            FAILURE_STRING.consistent.canBeParamPropsOnly,
+                            getFixerForLonghandProp(param, construct),
+                        );
+                }
+        }
+    }
+
+    private canBeParameterProperty(param: ts.ParameterDeclaration, construct: ts.ConstructorDeclaration): boolean {
+        const member = construct.parent!.members.find(
```

Comment:
Just a quick note, I don't have time for a full review now:
rest parameters need to be excluded as they are never allowed to be parameter properties